### ISSUE TITLE
requests.exceptions.HTTPError: 503 Server Error: Service Temporarily Unavailable for url: https://ega.ebi.ac.uk:8443/ega-openid-connect-server/token

### ISSUE THREAD

**Issue by sabrinacamp2** [EGA-archive/ega-download-client#168]
Hi all,

I am receiving an error when trying to list the datasets I have access to via the following command:
`pyega3 -cf credential_file.json -d datasets`

The error I receive after attempting to execute that command is the following:
```
Invalid username, password or secret key - please check and retry. If problem persists contact helpdesk on helpdesk@ega-archive.org
Traceback (most recent call last):
  File "/broad/VanAllenLab/xchip/cga_home/scamp/ega_download/lib/python3.10/site-packages/pyega3/libs/auth_client.py", line 37, in token
    r.raise_for_status()
  File "/broad/VanAllenLab/xchip/cga_home/scamp/ega_download/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 503 Server Error: Service Temporarily Unavailable for url: https://ega.ebi.ac.uk:8443/ega-openid-connect-server/token
```

Is there something going on with EGA's servers? Or are there any troubleshooting tips?

All the best,
Sabrina

**Comment by gh-cweiss** [comment]
I am having the same issue this morning while trying to list files in a dataset. 
`pyega3 -cf CREDENTIALS_FILE files EGAD00001002969`
I have had this working for the last several weeks and haven't changed anything about my credentials file or pyega3 installation.

**Comment by sabrinacamp2** [comment]
@gh-cweiss It's working for me today.

**Comment by gh-cweiss** [comment]
It is also working for me today.

**Comment by allaway** [comment]
I have been having this issue intermittently the past few days as well, including right now.

**Comment by sabrinacamp2** [comment]
No longer working for me

**Comment by CRMacPherson** [comment]
Same issue with me, even with just `-d -t datasets`. I thought the SSL certs might have expired but it has ~360 days left.

**Comment by allaway** [comment]
FYI, after reaching out to the EGA help desk they were willing to set up an
aspera workspace with the data I needed. It sounds like a server side issue
that they are not able to fix at the moment.

**Comment by CRMacPherson** [comment]
Thanks! I will ask. If it helps others I was able to resolve this temporarily by switching VPNs even though I could connect to the ports regardless.
But, the problem has returned.

**Comment by malloryfreeberg** [comment]
Hi all. I’m sorry you are experiencing this issue, and thank you for reporting it. We are aware of the instability of the service at the moment and are actively working to solve the underlying intermittent issues. If you continue to experience download issues, the best place to get assistance is by emailing the EGA Helpdesk ([helpdesk@ega-archive.org](mailto:helpdesk@ega-archive.org)) and attaching the output log produced by pyega3 (pyega3_output.log). The Helpdesk team can then provide further assistance and updates on a fix. Thank you in advance for your patience while we work to resolve this!

**Comment by malloryfreeberg** [comment]
Thank you, all, for your patience while we worked on and released a new version of both [pyega3 (v5.0.0)](https://pypi.org/project/pyega3/) and the underlying API. Please update to the newest version of pyega3 (you can find the instructions [here](https://github.com/EGA-archive/ega-download-client#installation-and-update)) to enjoy the new features, including a more stable service which results in a higher download success rate. There might still be periods of time where other factors affect the stability of our services, but we are always trying to improve upon these to provide a consistent service.

This GH issue will be closed, but if you continue to experience the same problem, feel free to re-open it. If you experience any other problems, please email the EGA Helpdesk at helpdesk@ega-archive.org and attach the output log produced by pyega3 (pyega3_output.log). The Helpdesk team can then provide further assistance. Thank you!
